examples/marching/src/host/paths.ads

```diff
@@ -0,0 +1,60 @@
+with Ada.Command_Line;
+with GNAT.OS_Lib;
+with Ada.Strings; use Ada.Strings;
+with Ada.Strings.Fixed; use Ada.Strings.Fixed;
+with Ada.Strings.Unbounded;
+
+package Paths is
+    type Path is new String;
+
+    function "+" (P : Path)
+      return String
+    is (String (P));
+
+    function "+" (S : Ada.Strings.Unbounded.Unbounded_String)
+      return Path
+    is (Path (Ada.Strings.Unbounded.To_String (S)));
+
+    function Is_Absolute (P : Path)
+      return Boolean is (GNAT.OS_Lib.Is_Absolute_Path (+P));
+
+    function Is_Directory (P : Path)
+      return Boolean is (GNAT.OS_Lib.Is_Directory (+P));
+
+    function Is_Regular_File (P : Path)
+      return Boolean is (GNAT.OS_Lib.Is_Regular_File (+P));
+
+    function Exists (P : Path)
+      return Boolean is (Is_Regular_File (P) or Is_Directory (P));
```

Comment:
I think you might be able to use [Ada.Directories.Exists](https://www.adaic.org/resources/add_content/standards/05rm/html/RM-A-16.html) there.